docs/rholang/rholangtut.md

```diff
@@ -11,44 +11,238 @@ There is not an IDE for Rholang. Get started with Rholang by selecting one of th
+
+2) Every name is of the form `@P`, where `P` is a rholang process.  The `!` production takes a name `n` on its left and a process `P` on its right, then sends `@P` over the channel named `n`.  Line 2 forms the name `@"Hello, world"` and sends it on the channel whose name is stored in the variable `HelloWorld`.
+
+3) This `for` production creates a process that waits for a single message to be sent on the channel whose name is stored in the variable `HelloWorld`.  The pattern `@text` gets matched against the serialized process, binding the process-valued variable `text` to the original process that was sent.
+
+4) The built-in name `@"stdout"` designates a channel on which scala code is listening.  The scala code prints out the process it receives.
+
+## Replicated receive
+
+    1 new HelloWorld in {
+    2   for (@text <= HelloWorld) {
+    3     @"stdout"!(text)
     4   } |
-    5   new myChannel in {
-    6     HelloWorld!(*myChannel)
-    7   }
-    8 }
+    5   HelloWorld!("Hello, world!") |
+    6   HelloWorld!("Hola, mundo!")
+    7 }
+
+2) Instead of handling only a single message, a `for` using a double arrow `<=` will persist, spawning a copy of the body for each message received.
+
+5-6) We send the string processes `"Hello, world!"` and `"Hola, mundo!"` on the channel `HelloWorld`.  It is non-deterministic which message will be processed first.
+
+## Contracts as sugar for replicated receive
+
+    1 new HelloWorld in {
+    2   contract HelloWorld(@text) = {
+    3     @"stdout"!(text)
+    4   } |
+    5   HelloWorld!("Hello, world!") |
+    6   HelloWorld!("Hola, mundo!")
+    7 }
+
+2) The only difference between this example and the last one is this line.  The `contract` production is syntactic sugar.  The process `contract Name(...formals) = { P }` means the same as `for (...formals <= Name) { P }`.
+
+## Replicated send
+
+    1 new HelloWorld in {
+    2   HelloWorld!!("Hello, world!") |
+    3   for (@text <- HelloWorld) {
+    4     @"stdout"!(text)
+    5   } |
+    6   for (@text <- HelloWorld) {
+    7     @"stderr"!(text)
+    8   }
+    9 }
+
+2) The double-bang `!!` means that this message will be sent again as soon as it is read.
+
+3-4, 6-7) There are two listening processes; each one consumes a single message from the channel and forwards it to either `"stdout"` or `"stderr"`.  The order in which they get forwarded to those channels is nondeterministic.
+
+## Sequential send
+
+In order to have one message follow after another is known to have been received, we must use an acknowledgement message.
+
+    1 new chan, ack in {
+    2   chan!(0) |
+    3   for (_ <- ack) {
+    4     chan!(1)
+    5   } |
+    6   for (@num <= chan) {
```

Comment:
I think that example, even though it's short, would be more readable if second `for (@num  <= chan) { … } ` part came before first one.